### DISCUSSION TITLE
Heterogeneous Enum

### DISCUSSION THREAD

**Discussion by Jack-Works** [Jack-Works/proposal-enum#8] [General]
In TypeScript, enum can be [Heterogeneous](https://www.typescriptlang.org/docs/handbook/enums.html#heterogeneous-enums). I very doubt if it is useful.

Flow.js does not allow heterogeneous enum.

```ts
enum BooleanLikeHeterogeneousEnum {
  No = 0,
  Yes = "YES",
}
```

## Problem

This conflicts with the limited computed value (#12) because we cannot specify symbols without ad-hoc syntax.

This conflicts with the inline-able requirement (#4) for the same reason.

Should we ban it?

**Comment by ljharb** [comment]
JS doesn’t have a strong type system. There is no good reason imo to ban mixed types of enums.
